library/modules/Textures.cpp

```diff
@@ -28,9 +30,36 @@ namespace DFHack {
 DBG_DECLARE(core, textures, DebugCategory::LINFO);
 }
 
+struct ReservedRange {
+    void init(int32_t start) {
+        this->start = start;
+        this->end = start + ReservedRange::size;
+        this->current = start;
+        this->is_installed = true;
+    }
+    long get_new_texpos() {
+        if (this->current == this->end)
+            return -1;
+        current = this->current;
```

Comment:
I don't see another `current` being declared anywhere. Isn't this line just assigning to itself?

Reply:
moreover, these three lines can be replaced by a single `return current++`